src/game/models/mod.rs

```diff
@@ -105,6 +144,15 @@ impl From<Size> for Vector {
     }
 }
 
+impl From<[f64; 2]> for Vector {
```

Comment:
I wonder if in a separate PR we should put `Vector` in it's own module. It's grown pretty big.